x-pack/legacy/plugins/code/server/lsp/workspace_handler.ts

```diff
@@ -315,18 +314,43 @@ export class WorkspaceHandler {
   private async resolveUri(uri: string) {
     if (uri.startsWith('git://')) {
       const { repoUri, file, revision } = parseLspUrl(uri)!;
-      const { workspaceRepo, workspaceRevision } = await this.openWorkspace(repoUri, revision);
+      let wRepo: Repository;
+      let wRevision: string;
+      try {
+        const { workspaceRepo, workspaceRevision } = await this.openWorkspace(repoUri, revision);
+        wRepo = workspaceRepo;
+        wRevision = workspaceRevision;
+      } catch (e) {
+        // If the language server return non head commit, like go langserver, swallow it.
+        // TODO(henrywong) Once kibana code support multiple revision, we should remove this catch statement.
+        let lang = '';
+        if (file) {
+          lang = await detectLanguage(file);
+        }
+        if (
+          Boom.isBoom(e) &&
+          (e as Error).message === 'revision must be master.' &&
+          lang === 'go'
```

Comment:
@spacedragon is this a go specific problem?

Reply:
@spacedragon ping.

Reply:
We should wait until we support multi-revision.